src/ProjectOrigin.Vault/CommandHandlers/CheckForWithdrawnCertificatesCommandHandler.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using MassTransit;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ProjectOrigin.Vault.Database;
+using ProjectOrigin.Vault.Models;
+using ProjectOrigin.Vault.Options;
+
+namespace ProjectOrigin.Vault.CommandHandlers;
+
+public record CheckForWithdrawnCertificatesCommand
+{
+}
+
+public class CheckForWithdrawnCertificatesCommandHandler : IConsumer<CheckForWithdrawnCertificatesCommand>
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly ILogger<CheckForWithdrawnCertificatesCommandHandler> _logger;
+    private readonly NetworkOptions _networkOptions;
+    public CheckForWithdrawnCertificatesCommandHandler(IUnitOfWork unitOfWork,
+        IOptions<NetworkOptions> networkOptions,
+        IHttpClientFactory httpClientFactory,
+        ILogger<CheckForWithdrawnCertificatesCommandHandler> logger)
+    {
+        _unitOfWork = unitOfWork;
+        _httpClientFactory = httpClientFactory;
+        _logger = logger;
+        _networkOptions = networkOptions.Value;
+    }
+
+    public async Task Consume(ConsumeContext<CheckForWithdrawnCertificatesCommand> context)
+    {
+        var stamps = _networkOptions.Stamps;
+
+        foreach (var stamp in stamps)
+        {
+            var cursors = await _unitOfWork.WithdrawnCursorRepository.GetWithdrawnCursors();
+            var matchingCursor = cursors.FirstOrDefault(x => x.StampName == stamp.Key) ?? new WithdrawnCursor
+            {
+                StampName = stamp.Key,
+                SyncPosition = 0,
+                LastSyncDate = DateTimeOffset.UtcNow
+            };
+
+            var client = _httpClientFactory.CreateClient();
```

Comment:
I think we should move Create client to be outside of for loop. Could use a lot of memory creating new HttpClients all the time.

Reply:
Problem is that the baseurl of the stamps differs. but memory wise they should override on each stamp loop. I added Dispose at the buttom.

Reply:
var client = _httpClientFactory.CreateClient(); <-- Your not supplying a URL, we can still reuse the same HttpClient and make many calls.

Reply:
corrected